inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/TableMetricStatementExecutor.java

```diff
@@ -0,0 +1,254 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.StandardCharsets;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * A {@link JdbcBatchStatementExecutor} that simply adds the records into batches of {@link
+ * java.sql.PreparedStatement} and doesn't buffer records in memory. Only used in Table/SQL API.
+ * Supported executors:TableBufferedStatementExecutor, TableBufferReducedStatementExecutor, TableSimpleStatementExecutor
+ */
+public final class TableMetricStatementExecutor implements JdbcBatchStatementExecutor<RowData> {
+
+    private static final Pattern pattern = Pattern.compile("Batch entry (\\d+)");
+    private static final Logger LOG = LoggerFactory.getLogger(TableMetricStatementExecutor.class);
+    private final StatementFactory stmtFactory;
+    private final JdbcRowConverter converter;
+    private List<RowData> batch;
+    private final DirtySinkHelper<Object> dirtySinkHelper;
+    private final SinkMetricData sinkMetricData;
+    private final AtomicInteger counter = new AtomicInteger();
+    private transient FieldNamedPreparedStatement st;
+    private boolean multipleSink;
+    private String label;
+    private String logtag;
+    private String identifier;
+    private Function<RowData, RowData> valueTransform = null;
+    // counters used for table level metric calculation for multiple sink
+    public long[] metric = new long[4];
+
+    public TableMetricStatementExecutor(StatementFactory stmtFactory, JdbcRowConverter converter,
+            DirtySinkHelper<Object> dirtySinkHelper, SinkMetricData sinkMetricData) {
+        this.stmtFactory = checkNotNull(stmtFactory);
+        this.converter = checkNotNull(converter);
+        this.batch = new CopyOnWriteArrayList<>();
+        this.dirtySinkHelper = dirtySinkHelper;
+        this.sinkMetricData = sinkMetricData;
+    }
+
+    public void setDirtyMetaData(String label, String logtag, String identifier) {
+        this.label = label;
+        this.logtag = logtag;
+        this.identifier = identifier;
+    }
+
+    public void setMultipleSink(boolean multipleSink) {
+        this.multipleSink = multipleSink;
+    }
+
+    @Override
+    public void prepareStatements(Connection connection) throws SQLException {
+        st = stmtFactory.createStatement(connection);
+    }
+
+    public void setValueTransform(Function<RowData, RowData> valueTransform) {
+        this.valueTransform = valueTransform;
+    }
+
+    @Override
+    public void addToBatch(RowData record) throws SQLException {
+        if (multipleSink && record.getRowKind().equals(RowKind.UPDATE_BEFORE)) {
```

Comment:
used to fix the metric for update_before

Reply:
rowkind all is INSERT when multipleSink.